i18n(ja): Add new guides/deploy/netlify.mdx translation (#4611)
withastro/docs: src/content/docs/ja/guides/deploy/netlify.mdx

---
title: AstroサイトをNetlifyにデプロイする
description: NetlifyでAstroサイトをウェブにデプロイする方法。
type: deploy
i18nReady: true
---
[Netlify](https://netlify.com)は、ウェブアプリケーションや静的ウェブサイトのためのホスティングとサーバーレスバックエンドサービスを提供しています。Astroサイトは全てNetlifyでホストできます！

このガイドは、NetlifyのウェブサイトUIまたはNetlifyのCLIを介してデプロイするための手順を説明しています。

> ## プロジェクトの設定
> 
> Astoプロジェクトは、静的サイト、サーバーレンダリングサイト、(実験的な)エッジレンダリングサイトの3つの異なる方法でNetlifyにデプロイできます。
> 
> ### 静的サイト
> 
> Astoプロジェクトは、デフォルトでは静的サイトです。Netlifyに静的Astroサイトをデプロイするために、特に追加の設定は必要ありません。

Reviewer: タイポしているようです

```suggestion
## プロジェクトの設定

Astroプロジェクトは、静的サイト、サーバーレンダリングサイト、(実験的な)エッジレンダリングサイトの3つの異なる方法でNetlifyにデプロイできます。

### 静的サイト

Astroプロジェクトは、デフォルトでは静的サイトです。Netlifyに静的Astroサイトをデプロイするために、特に追加の設定は必要ありません。
```

Author: 確認ありがとうございます！修正しました...！

### SSR/Edge用のアダプターを追加する

> AstroプロジェクトにSSRを有効にし、Netlifyにデプロイするには、以下の`astro add`コマンドで[Netlifyアダプター](/en/guides/integrations-guide/netlify/)を追加します。これにより、アダプターがインストールされ、`astro.config.mjs`ファイルが1つのステップで適切に変更されます。

Author: [Netlifyアダプター](/en/guides/integrations-guide/netlify/) は日本語訳されたドキュメントがまだないので、一時的に英語のドキュメントのURLにしています。

```bash
npx astro add netlify
```

> アダプターを手動でインストールする場合は、以下の2つのステップでインストールができます。:

Reviewer 2: こちらも @ryuapp さんのご指摘どおりコロンは不要です。あと、「インストール」という言葉が一文に二回出てきており少し冗長なため、以下のようにするのはどうでしょうか:

```suggestion
アダプターを手動でインストールする場合は、以下の2つのステップを実行してください。
```

> 1. [`@astrojs/netlify` アダプター](https://github.com/withastro/astro/tree/main/packages/integrations/netlify)を、パッケージマネージャーを使ってプロジェクトのdependenciesにインストールします。npmをパッケージマネージャーとして使用しているかどうかわからない場合は、ターミナルで以下のコマンドを実行します。:

Reviewer 2: dependencies は他のページでは「依存関係」と訳していることが多いため、@ryuapp さんの指摘に加えてそれも反映すると良さそうです:

```suggestion
1. [`@astrojs/netlify`アダプター](https://github.com/withastro/astro/tree/main/packages/integrations/netlify)を、パッケージマネージャーを使ってプロジェクトの依存関係にインストールします。npmをパッケージマネージャーとして使用しているかどうかわからない場合は、ターミナルで以下のコマンドを実行します。
```

    ```bash
      npm install @astrojs/netlify
    ```

1. `astro.config.mjs`プロジェクト設定ファイルに新しく以下の2行を追加します。

    ```js title="astro.config.mjs" ins={2, 5-6}
    import { defineConfig } from 'astro/config';
    import netlify from '@astrojs/netlify/functions';

    export default defineConfig({
      output: 'server',
      adapter: netlify(),
    });
    ```

    [Netlifyの実験的Edge Functions](https://docs.netlify.com/netlify-labs/experimental-features/edge-functions/#app)を使ってプロジェクトをレンダリングするには、Astro設定ファイルの`netlify/functions`インポートを`netlify/edge-functions`に変更します。
      ```js title="astro.config.mjs" ins={3} del={2}
      import { defineConfig } from 'astro/config';
      import netlify from '@astrojs/netlify/functions';
      import netlify from '@astrojs/netlify/edge-functions';

      export default defineConfig({
        output: 'server',
        adapter: netlify(),
      });
      ```

## デプロイ方法

> NetlifyのウェブサイトUI、もしくはNetlifyのCLI(コマンドラインインターフェース)を使うことでデプロイすることができます。デプロイする方法は、静的サイトもSSRサイトも同じ方法です。

Reviewer: 翻訳ガイドで「冗長表現は避ける必要がある」とあるので、簡単に訳すと良いと思います
> - 冗長表現はさけてください
>   - することができます → できます
>   - 使うことができます → 使えます

```suggestion
NetlifyのウェブサイトUI、もしくはNetlifyのCLI(コマンドラインインターフェース)を使うことでデプロイできます。デプロイする方法は、静的サイトもSSRサイトも同じ方法です。
```

Reviewer 2: この行の二つ目の文について気付いたのですが、「方法は、〜も同じ方法です」と「方法」が二回出てきてしまっているので、@ryuapp さんの指摘に加えて調整した方が良さそうです:

```suggestion
NetlifyのウェブサイトUI、もしくはNetlifyのCLI(コマンドラインインターフェース)を使うことでデプロイできます。デプロイする方法は、静的サイトもSSRサイトも同じです。
```

### ウェブサイトUIを使ったデプロイ方法

GitHub、GitLab、BitBucket、またはAzure DevOpsにプロジェクトがある場合は、NetlifyのウェブサイトUIを使用してAstroサイトをデプロイできます。

1. [Netlifyダッシュボード](https://app.netlify.com/)で<kbd>Add a new site</kbd>をクリックします。

2. <kbd>Import an existing project</kbd>を選択します。

    GitプロバイダーからAstroリポジトリをインポートする場合、Netlifyが自動で正しい構成設定を事前に入力します。

3. 以下の設定が入力されていることを確認し、<kbd>Deploy</kbd>ボタンをクリックします。:

    - **Build Command:** `astro build` もしくは `npm run build`
    - **Publish directory:** `dist`

>  デプロイが完了すると、サイトの概要ページにリダイレクトされます。そこでサイトの設定を編集することができます。

Reviewer: 冗長表現を解消すると良いと思います
```suggestion
 デプロイが完了すると、サイトの概要ページにリダイレクトされます。そこでサイトの設定を編集できます。
```

ソースリポジトリ内の変更があった際には、デプロイ設定に基づいてプレビューと本番環境のデプロイをトリガーします。

#### `netlify.toml` ファイル

> オプションとして、プロジェクトリポジトリのトップレベルに新しく `netlify.toml` ファイルを作成し、ビルドコマンドと公開ディレクトリ、環境変数やリダイレクトなどのプロジェクトの設定をすることができます。Netlifyはこのファイルを読み込み、デプロイを自動で設定します。

Reviewer: 冗長表現を解消すると良いと思います
```suggestion
オプションとして、プロジェクトリポジトリのトップレベルに新しく `netlify.toml` ファイルを作成し、ビルドコマンドと公開ディレクトリ、環境変数やリダイレクトなどのプロジェクトの設定ができます。Netlifyはこのファイルを読み込み、デプロイを自動で設定します。
```

Reviewer 2: netlify.toml 前後のスペースも取り除いてあげたほうが良さそうです:

```suggestion
オプションとして、プロジェクトリポジトリのトップレベルに新しく`netlify.toml`ファイルを作成し、ビルドコマンドと公開ディレクトリ、環境変数やリダイレクトなどのプロジェクトの設定ができます。Netlifyはこのファイルを読み込み、デプロイを自動で設定します。
```

デフォルトの設定を構成するには、以下の`netlify.toml`ファイルを作成します。:

```toml
[build]
  command = "npm run build"
  publish = "dist"
```

📚 さらに詳しい情報は、Netlifyのブログの[“Deploying an existing Astro Git repository”](https://www.netlify.com/blog/how-to-deploy-astro/#deploy-an-existing-git-repository-to-netlify)を参照してください。

### CLIを使ったデプロイ方法

[Netlify CLI](https://cli.netlify.com/)を使って新しくNetlifyサイトを作成し、Gitリポジトリをリンクすることもできます。

1. グローバルにNetlifyのCLIをインストールします。

    ```bash
    npm install --global netlify-cli
    ```

> 2. `netlify login` を実行し、Netlifyにログインして認証します。
> 3. `netlify init` を実行し、手順に沿って進めます。

Reviewer: バッククオートの後の空白は、不要だと思います
```suggestion
2. `netlify login`を実行し、Netlifyにログインして認証します。
3. `netlify init`を実行し、手順に沿って進めます。
```

> 4. (`astro build`)のビルドコマンドで確認します。

Reviewer 2: 原文は

> Confirm your build command (astro build)

なのと、「括弧 + の」となっているのが少し違和感があるため、

```suggestion
4. ビルドコマンド(`astro build`)を確認します。
```

のようにするのはどうでしょうか（原文も含め若干舌っ足らず感がありますが...）。

    CLIはビルドコマンド(`astro build`)とデプロイディレクトリ(`dist`)を自動で検出し、それらの設定を含んだ[`netlify.toml`ファイル](#netlifytoml-file)を自動で生成することを選択するかどうか聞きます。

5.  Gitにプッシュしてビルドとデプロイを行います。

    CLIはリポジトリにデプロイキーを追加します。そのため、`git push`するたびにNetlifyでサイトが自動で再ビルドされます。

> 📚 さらに詳しい情報は、Netlifyのブログの[Deploy an Astro site using the Netlify CLI](https://www.netlify.com/blog/how-to-deploy-astro/#link-your-astro-project-and-deploy-using-the-netlify-cli)を参照してください。

Reviewer: 訳しても良いと思います
```suggestion
📚 さらに詳しい情報は、Netlifyのブログの[Netlify CLIを使ったAstroサイトのデプロイ](https://www.netlify.com/blog/how-to-deploy-astro/#link-your-astro-project-and-deploy-using-the-netlify-cli)を参照してください。
```

### Node.jsバージョンの設定

Netlifyのレガシー機能[build image](https://docs.netlify.com/configure-builds/get-started/#build-image-selection) (Xenial)を使用している場合は、Node.jsの正しいバージョンが設定されていることを確認してください。AstroはNode.js`v18.14.1`以上のバージョンが必要です。

> 以下の方法で[Node.jsのバージョンをNetlify上で指定](https://docs.netlify.com/configure-builds/manage-dependencies/#node-js-and-javascript)することができます。:

Reviewer: 冗長表現の解消と「：」を削除すると良いと思います
```suggestion
以下の方法で[Node.jsのバージョンをNetlify上で指定](https://docs.netlify.com/configure-builds/manage-dependencies/#node-js-and-javascript)できます。
```

> - ベースディレクトリ上にある[`.nvmrc`](https://github.com/nvm-sh/nvm#nvmrc)ファイル。
> - Netlifyプロジェクトダッシュボード上にある、サイト設定`NODE_VERSION`環境変数。

Reviewer: 箇条書きの部分なので、「。」はない方が見栄えが良いと思います
```suggestion
- ベースディレクトリ上にある[`.nvmrc`](https://github.com/nvm-sh/nvm#nvmrc)ファイル
- Netlifyプロジェクトダッシュボード上にある、サイト設定`NODE_VERSION`環境変数
```

## Netlify Functionsを使用する方法

Netlify FunctionsとAstroを併用して使う場合、特別な設定は必要ありません。プロジェクトルートに`netlify/functions`ディレクトリを追加し、[Netlify Functionsのドキュメント](https://docs.netlify.com/functions/overview/)に従って進めてください！

## 例

> - [How to deploy an Astro site](https://www.netlify.com/blog/how-to-deploy-astro/) — Netlifyブログ
> - [Deploy An Astro site with Forms, Serverless Functions, and Redirects](https://www.netlify.com/blog/deploy-an-astro-site-with-forms-serverless-functions-and-redirects/) — Netlifyブログ
> - [Deployment Walkthrough Video](https://youtu.be/GrSLYq6ZTes) — Netlify YouTubeチャンネル

Reviewer: 訳しても良いと思います(もっと良い翻訳があればそちらにして頂ければと思います)

```suggestion
- [Astroサイトのデプロイ方法](https://www.netlify.com/blog/how-to-deploy-astro/) — Netlifyブログ
- [フォーム、サーバーレスファンクション、リダイレクトを利用したAstroサイトのデプロイ](https://www.netlify.com/blog/deploy-an-astro-site-with-forms-serverless-functions-and-redirects/) — Netlifyブログ
- [開発の実演動画](https://youtu.be/GrSLYq6ZTes) — Netlify YouTubeチャンネル
```
